### PR TITLE
YAPF'd, isort'd, changelog'd: 0.7.0 Release

### PR DESCRIPTION
**Pull request by Lnaden**
Code change: I made a `AutodocBaseSettings` class which uses the `PydanticAutodocMeta` metaclass. This is not a formal `Protomodel` as it is not something which will be serialized through msgpack and is really meant to be used as a settings class rather than a data model. Although Elemental does not use this, Fractal will (e.g. CLI-parsing classes). 

YAPF has been applied and also told to ignore certain blocks structured for our clarity. I have also added a `.yapfignore` file to ignore whole auto-generated files (can't do this in the `setup.cfg` `[yapf]` section from what I can tell).

isort has been applied and also told to ignore the auto-generated files through its `setup.cfg` header.

Changelog updated for the 0.7.0 minor bump release.

### CODE REVIEW

**Comment by Lnaden** [comment]
I still need to make a proper LGTM cleanup, but figured this should go in first.

File: nist_data/build_period_table.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 import re
```
Comment (loriab) [inline nist_data/build_period_table.py]: request "ic" added to file name

File: qcelemental/__init__.py [hunk 2]
```diff
@@ -2,24 +2,21 @@
 Main init for QCElemental
 """
 
+from . import models, molparse, molutil, util
```
Comment (loriab) [inline qcelemental/__init__.py]: I didn't know you could do this

File: qcelemental/molparse/from_arrays.py [hunk 3]
```diff
@@ -353,37 +353,34 @@ def from_arrays(*,
...
-            zero_ghost_fragments=zero_ghost_fragments,
-            verbose=verbose)
+        processed = validate_and_fill_chgmult(zeff=Z_available,
+                                              fragment_separators=molinit['fragment_separators'],
+                                              molecular_charge=molecular_charge,
+                                              fragment_charges=molinit['fragment_charges'],
+                                              molecular_multiplicity=molecular_multiplicity,
+                                              fragment_multiplicities=molinit['fragment_multiplicities'],
+                                              zero_ghost_fragments=zero_ghost_fragments,
+                                              verbose=verbose)
```
Comment (loriab) [inline qcelemental/molparse/from_arrays.py]: @Lnaden check just once that your yapf is up-to-date. Daniel and I had dueling yapfs for a while.

Comment (Lnaden) [reply qcelemental/molparse/from_arrays.py]: I just installed/updated from c-f before doing this, so I should be good.

File: nist_data/build_period_table.py (hunk 1 above)
Comment (Lnaden) [reply nist_data/build_period_table.py]: Will do!

**Pull request merged**